app/services/album.js

```diff
@@ -25,3 +26,41 @@ exports.photos = albumId =>
     .catch(() => {
       throw errors.externalApiError('Error in external API');
     });
+
+exports.buyAlbum = async (albumId, currentUser) => {
+  const response = await axios.get(`${apiUrl}/albums/${albumId}`).catch(() => {
+    throw errors.databaseError();
+  });
+  let album = await models.album.findOne({ where: { title: response.data.title } }).catch(() => {
+    throw errors.databaseError();
+  });
+
+  if (!album) {
+    album = await models.album
+      .create({
+        title: response.data.title,
+        userId: currentUser.dataValues.id
+      })
+      .catch(() => {
+        throw errors.databaseError();
+      });
+  }
+
+  const boughtAlbum = await models.userAlbums
+    .findOne({
+      where: { albumId: album.id, userId: currentUser.id }
+    })
+    .catch(() => {
+      throw errors.databaseError();
+    });
+
+  if (boughtAlbum) {
+    throw errors.albumPurchased();
+  }
+
+  await currentUser.addAlbum(album).catch(() => {
+    throw errors.databaseError();
```

Comment:
You have a lot of generic `databaseError` here. If something fails in a production environment, you'd have no way of distinguishing where it failed. The `databaseError()` function can take descriptions, so add them to better differentiate between them.